Client/Frontend/Autofill/CreditCard/CreditCardBottomSheet/CreditCardBottomSheetViewModel.swift

```diff
@@ -161,25 +179,61 @@ struct CreditCardBottomSheetViewModel {
                                                                        with: ccNumberDecrypted,
                                                                        fieldValues: decryptedCreditCard)
             return updatedDecryptedCreditCard
+        case .selectSavedCard:
+            guard let row = row,
+                  let selectedCreditCard = getSavedCreditCard(for: row)
+            else {
+                return nil
+            }
+
+            let decryptedCreditCardNum = decryptCreditCardNumber(card: selectedCreditCard)
+
+            guard !decryptedCreditCardNum.isEmpty else {
+                return nil
+            }
+
+            let plainTextCard = UnencryptedCreditCardFields(ccName: selectedCreditCard.ccName,
+                                                            ccNumber: decryptedCreditCardNum,
+                                                            ccNumberLast4: selectedCreditCard.ccNumberLast4,
+                                                            ccExpMonth: selectedCreditCard.ccExpMonth,
+                                                            ccExpYear: selectedCreditCard.ccExpYear,
+                                                            ccType: selectedCreditCard.ccType)
+            return plainTextCard
         }
     }
 
     func getConvertedCreditCardValues(bottomSheetState: CreditCardBottomSheetState,
-                                      ccNumberDecrypted: String) -> CreditCard? {
+                                      ccNumberDecrypted: String,
+                                      row: Int? = nil) -> CreditCard? {
```

Comment:
💼 TODO: Update test

Reply:
Same here, it has test for both save and update but I added a ticket to add more tests for selected card as that involves decryption which comes from rust autofill.
[JIRA](https://mozilla-hub.atlassian.net/browse/FXIOS-6719)